### PR TITLE
deprecating `jarowinkler` in favor of `jaro_winkler`

### PR DESCRIPTION
**Pull request by olcbean**
`Jaro-Winkler` distance alg is following the naming convention used for `Damerau–Levenshtein` as per [comment](https://github.com/elastic/elasticsearch/pull/27409#discussion_r151834860)

### CODE REVIEW

**Comment by dnhatn** [comment]
@elasticmachine please test this.

File: core/src/main/java/org/elasticsearch/search/suggest/phrase/DirectCandidateGeneratorBuilder.java [hunk 1]
```diff
@@ -474,9 +474,7 @@ static StringDistance resolveDistance(String distanceVal) {
             return new LevensteinDistance();
         } else if ("levenshtein".equals(distanceVal)) {
             return new LevensteinDistance();
-            // TODO Jaro and Winkler are 2 people - so apply same naming logic
-            // as damerau_levenshtein
-        } else if ("jarowinkler".equals(distanceVal)) {
+        } else if ("jaro_winkler".equals(distanceVal) || "jarowinkler".equals(distanceVal)) {
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/search/suggest/phrase/DirectCandidateGeneratorBuilder.java]: Do we really need to support both? Can we settle on one (with the underscore), and deprecate the other? And same for `damerau_levenshtein`?

**Review by jasontedor — CHANGES_REQUESTED** [review]
I left some questions.

File: core/src/main/java/org/elasticsearch/search/suggest/phrase/DirectCandidateGeneratorBuilder.java (hunk 1 above)
Comment (DaveCTurner) [reply core/src/main/java/org/elasticsearch/search/suggest/phrase/DirectCandidateGeneratorBuilder.java]: +1 for deprecation of `jarowinkler`.

Not sure about that score creep tho :)

Comment (jasontedor) [reply core/src/main/java/org/elasticsearch/search/suggest/phrase/DirectCandidateGeneratorBuilder.java]: Agree, the other should be separate.

**Comment by olcbean** [comment]
I just deprecated `jarowinkler` and rebased.

**Comment by jasontedor** [comment]
retest this please

**Review by jasontedor — APPROVED** [review]
LGTM.

**Comment by jasontedor** [comment]
@DaveCTurner Did you want to give this one a review too?

**Review by DaveCTurner — APPROVED** [review]
LGTM too.

**Comment by DaveCTurner** [comment]
Merged and backported to `6.x` in 1eb5cb1816

**Comment by jasontedor** [comment]
Thanks @DaveCTurner.

**Pull request merged**